packages/blaze/view.js

```diff
@@ -352,13 +352,16 @@ Blaze._materializeView = function (view, parentView, _workStack, _intoArray) {
       view._isInRender = false;
 
       if (! c.firstRun) {
+        // added by ramezrafla@zegenie to skip affecting DOM for unchanged content
+        if (Blaze._deepCompare(lastHtmljs,htmljs)) {
```

Comment:
> But I did see somewhere in the code that htmljs can be views. Also, we skipped DOM comparison altogether that was there before, which was my biggest concern. If I was too cautious, then we can simply simplify the compare function.

Yes, but at a higher level. These views which have `!computation.firstRun` seems to be very simple ones.

Which is also a problem probably, that higher-levels are recreated.

So can you make in your codebase (on which you are doing these optimizations here) a check to print out when `computation.firstRun` is false and `htmljs` is something complicated.

> But, why would the original developers convert to DOM if it was always primitives?

I have no idea why they convert at all before the `if`. The fact that you moved `if` out is great, but I have no idea why they did it like this. Does converting to DOM has some side-effects we do not know?